pkg/plugin/processor/builtin/webhook/http.go

```diff
@@ -0,0 +1,307 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate paramgen -output=http_paramgen.go httpConfig
+
+package webhook
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/conduitio/conduit-commons/opencdc"
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/goccy/go-json"
+	"github.com/jpillora/backoff"
+)
+
+type httpConfig struct {
+	// URL used in the HTTP request.
+	URL string `json:"request.url" validate:"required"`
+	// Method is the HTTP request method to be used.
+	Method string `json:"request.method" default:"POST"`
+	// ContentType is the value of the Content-Type header.
+	ContentType string `json:"request.contentType" default:"application/json"`
+
+	// BackoffRetryCount is the maximum number of retries for an individual record
+	// when backing off following an error.
```

Comment:
Given that these comments will now appear in the documentation 1:1 I'm wondering if we should ignore the Go convention of starting with the field name and rather just write the documentation, like:
```suggestion
	// The maximum number of retries for an individual record
	// when backing off following an error.
```

Reply:
That makes sense (even though I think it doesn't look too bad even with the field name). I've added a todo for us in https://github.com/ConduitIO/conduit/issues/1290, so we can go through other processors as well.